File: Programmers/Level2/N개의최소공배수/N개의최소공배수.js
function lcs(a, b) {
  if (!(a % b)) return b;
  return lcs(b, a % b);
}

function solution(arr) {
  let answer = 1;
  for (let i = 0; i < arr.length; i++) {
    answer = parseInt((answer * arr[i]) / lcs(answer, arr[i]));
  }
  return answer;
}

const arr = [2, 6, 8, 14];
console.log(solution(arr));

File: Programmers/Level2/N개의최소공배수/N개의최소공배수.py
def lcs(a, b):
    if a % b == 0: 
        return b 
    else:
        return lcs(b, (a % b))

def solution(arr):
    answer = 1 
    for num in arr:
        answer = (answer * num) // lcs(answer, num)
    return answer

arr = [2,6,8,14]
print(solution(arr))